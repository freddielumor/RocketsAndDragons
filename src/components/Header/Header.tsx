import * as React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { media } from '../../styles/mediaQueries'
import { appStyles } from '../../styles/appStyles'
import logoImage from '../../images/rocket.svg'

type IHeaderProps = {
  children: React.ReactNode
}

export const AppHeader: React.FC<IHeaderProps> = ({ children }) => (
  <Header>
    <Link to="/">
      <Logo src={logoImage} alt="Logo" />
      <Title>Rockets & Dragons</Title>
    </Link>
    <Divider />

    <Content>{children}</Content>
  </Header>
)

/* -- Components -- */

// Header
export const Header = styled.header`
  text-align: center;
`

// Header Content
const Content = styled.div``

// Logo
export const Logo = styled.img`
  max-width: 3.125rem;

  @media ${media.tablet} {
    max-width: 5rem;
  }
`

// Title
export const Title = styled.h1`
  color: ${appStyles.color.primary};
  font-size: 1.2rem;
  font-family: ${appStyles.font.extraBold};
  text-transform: uppercase;

  @media ${media.mobile} {
    font-size: 1.8rem;
  }

  @media ${media.tablet} {
    font-size: 2.5rem;
  }
`

// Divider
export const Divider = styled.hr`
  background-color: ${appStyles.color.primary};
  border: 1px solid ${appStyles.color.primary};
  height: 0.05rem;
  margin: 1rem 0;
`
