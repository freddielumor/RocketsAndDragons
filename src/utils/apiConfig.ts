// Congigure Endpoints For Data Fetching
const BASE_URL = 'https://api.spacexdata.com/v3'

export const ROCKETS_DATA_ENDPOINT = `${BASE_URL}/rockets`
export const DRAGONS_DATA_ENDPOINT = `${BASE_URL}/dragons`
