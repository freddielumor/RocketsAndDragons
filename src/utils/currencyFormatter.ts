// Format currency helper
export const formatCurrency = (value: number): string => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
  }).format(value)
}
