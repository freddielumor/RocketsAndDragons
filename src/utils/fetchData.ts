import axios from 'axios'

// Data Fetching Utility
export default async function fetchData(url: string): Promise<any> {
  try {
    const response = await axios.get(url)
    return response
  } catch (error) {
    return error
  }
}
