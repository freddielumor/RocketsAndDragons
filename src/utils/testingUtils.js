import * as React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { createStore, applyMiddleware } from 'redux'
import { Provider as ReduxProvider } from 'react-redux'
import thunk from 'redux-thunk'
import { render } from '@testing-library/react'
import rootReducer from '../redux/reducers/index'
import '@testing-library/jest-dom'

// TESTING UTILITY

// This util alllows us to replicate the application state for testing purposes.
// We export a render function which contains the application state, to allow us to test our components in isolation their integration across the app.

// 1. Setup initial state and mock store
const initialState = {}
const mockStore = createStore(rootReducer, initialState, applyMiddleware(thunk))

// 2. Create a test provider containing the state & router
const testWrapper = ({ children }) => {
  return (
    <Router>
      <ReduxProvider store={mockStore}>{children}</ReduxProvider>
    </Router>
  )
}

// 3. Create the test render function which is used in our test files
const testRender = (ui) => render(ui, { wrapper: testWrapper })

// 4. re-export everything from Testing Library
export * from '@testing-library/react'

// 5. Override render method
export { testRender as render }
