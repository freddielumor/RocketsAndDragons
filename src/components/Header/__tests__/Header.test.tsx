import * as React from 'react'
import { render } from '../../../utils/testingUtils'
import 'jest-styled-components'

import { Header, Title } from '../Header'

// Header
it('renders Header correctly', () => {
  const { container } = render(
    <Header>
      <div />
    </Header>,
  )

  expect(container).toMatchSnapshot()
})

// Title
describe('<Title />', () => {
  test('should render title', async () => {
    const props = {
      title: 'Test Title',
    }

    const { getByText } = render(<Title>{props.title}</Title>)

    getByText('Test Title')
  })
})
