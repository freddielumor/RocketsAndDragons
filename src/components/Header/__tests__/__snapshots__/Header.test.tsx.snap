// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
.c0 {
  text-align: center;
}

<div>
  <header
    class="c0"
  >
    <div />
  </header>
</div>
`;
